datafusion/optimizer/src/analyzer/rewrite_expr.rs

```diff
@@ -137,6 +136,19 @@ impl TreeNodeRewriter for OperatorToFunctionRewriter {
         }) = expr
         {
             match field {
+                GetFieldAccess::NamedStructField { name, .. } => {
+                    let expr = *expr.clone();
+                    let name = name.clone();
+                    let args = vec![expr, Expr::Literal(name)];
+                    return Ok(Transformed::yes(Expr::ScalarFunction(
+                        ScalarFunction::new_udf(
+                            Arc::new(ScalarUDF::new_from_impl(
+                                datafusion_functions::core::getfield::GetFieldFunc::new(),
+                            )),
+                            args,
+                        ),
+                    )));
+                }
```

Comment:
I know the method from https://github.com/apache/arrow-datafusion/pull/9546#discussion_r1519548725, but it seems that here it can't get the `context_provider`?

Reply:
only sql crate can get it directly

Reply:
I think we could implement this rewrite more easily using the API proposed in https://github.com/apache/arrow-datafusion/pull/9583 (so you could use datafusion_functions directly)